src/status_im/navigation/effects.cljs

```diff
@@ -77,17 +77,17 @@
 (defn- navigate
   [[component theme animations]]
   (let [{:keys [options]} (get views/screens component)
-        options           (if (map? animations) (assoc options :animations animations) options)]
-    (dismiss-all-modals)
+        options           (if (map? animations) (assoc options :animations animations) options)
+        parent            (or (first @state/modals) @state/root-id)]
     (navigation/push
-     (name @state/root-id)
+     (name parent)
```

Comment:
Onboarding navigation is very confusing because:
- We use the same screens across different flows (e.g., create an account, sign in, profiles → create an account).
- Since we use the base screen as a modal for transparency, and then push screens using `navigate-to-within-stack`, we have to track where we are coming from. Additionally, because we also use `navigate-to-within-stack` in events, we end up polluting the app-db with flow-related data to keep track of this.

**Solution:** In onboarding, we push stacks inside the base modal. Wallet likely does something similar, with different logic for where to pop up (though I'm not sure — I need to check).

- We continue using `navigate-to` also for these flows, as we do in other areas. Navigation will push into the modal if it exists; otherwise, it will push into the base stack.
- Do not change `navigate-to`; instead, create a new event, `navigate-within-base-modal`.

In both cases, we will eliminate confusion and reduce noisy code, as seen in https://github.com/status-im/status-mobile/pull/21668/commits/336f1cb7a2f4b31fab2d530b76e7b663d56b4443. (still WIP)

Reply:
i would say we should go a correct and simplest way, and don't use the base screen as a modal for transparency, these are should be a normal screens, and we should reuse same screens across different flows, or at least screens shouldn't have conditions inside

Reply:
navigation inside modal screen is a wrong itself

Reply:
Thank you @flexsurfer for your input. I agree we should not implement the whole onboarding flow inside a modal. I will try to address this in refactor PR and see how it looks.

Reply:
Hi @ilmotta,

Do we all agree on this? The removal of the modal-based flow for onboarding. Should I go ahead with removal, or do we need to discuss it further?

Reply:
Yes @Parveshdhull, I agree with @flexsurfer suggestion https://github.com/status-im/status-mobile/pull/21655#discussion_r1857905174 is the way to go.